In [135]:
import pandas as pd
from datetime import datetime

In [124]:
# Giving column names according to our wish, and reading the excel file
user_cols = ['Invoice_number', 'Date', 'Customer_name', 'Place', 'Product', 'Quantity','Serial','Amount','Tax','Contact_number']
orders = pd.read_excel('JUNE  2017-18.xls', names = user_cols, skiprows = 5, header = None)
orders.head()   

Invoice_number      Date Customer_name   Place            Product Quantity  \
0            NaN       NaN           NaN     NaN                NaN      NaN   
1           3090  05.06.17   MM Forgings  Guindy  Inter link cables      1No   
2            NaN       NaN           NaN     NaN                NaN      NaN   
3           3092  05.06.17   MM Forgings  Guindy      100 Ah B Rack      1No   
4            NaN       NaN           NaN     NaN                NaN      NaN   

  Serial  Amount  Tax Contact_number  
0    NaN     NaN  NaN            NaN  
1    NaN     NaN  NaN       22352518  
2    NaN     NaN  NaN            NaN  
3    NaN     NaN  NaN     9840980468  
4    NaN     NaN  NaN            NaN

In [125]:
# dropping the columns Serial, Amount and Tax as almost all entries are of 'NaN'

orders.dropna(how = 'all', inplace = True)
orders.drop(orders.columns[[6, 7, 8]], axis = 1, inplace = True)
orders

Invoice_number      Date                           Customer_name  \
1             3090  05.06.17                             MM Forgings   
3             3092  05.06.17                             MM Forgings   
5             3093  05.06.17                           Valli Textile   
7             3099  07.06.17                             Tafe Access   
9             3101  09.06.17                     Sysveda Information   
11            3102  10.06.17                            Brakes India   
12             NaN       NaN                                     NaN   
14            3113  15.06.17                         Tekvision Voice   
16            3114  19.06.17                          Easy Solutions   
18            3117  20.06.17                             MM Forgings   
19           4189/  20.06.17                             MM Forgings   
21           3121/  22.06.17                         Are Pee Polymer   
22             NaN       NaN                                     NaN   
24            3124  23.06.17                      ITC Ltd My Fortune   
26            3125  23.06.17                               Wave Tech   
28            3133  27.06.17                         RS Technologies   
30            3140  29.06.17                               Medopharm   
31             NaN       NaN                                     NaN   
32             NaN       NaN                                     NaN   
34            3145  30.06.17                         Tekvision Voice   
36            2091  01.06.17                            USUI Susira    
37             NaN       NaN                                     NaN   
38             NaN       NaN                                     NaN   
40            2092  01.06.17                            USUI Susira    
41             NaN       NaN                                     NaN   
42             NaN       NaN                                     NaN   
44            2093  01.06.17                            USUI Susira    
45             NaN       NaN                                     NaN   
46             NaN       NaN                                     NaN   
48            2094  01.06.17                      Technical Services   
..             ...       ...                                     ...   
221            NaN       NaN                       St.Andrews Church   
222            NaN       NaN                                     NaN   
224           2151  28.06.17                             MM Forgings   
225            NaN       NaN                                     NaN   
226            NaN       NaN                                     NaN   
228           2152  29.06.17                     Vivekananda Medical   
229            NaN       NaN                                     NaN   
231           2153  29.06.17                   Swaminathan Enterprie   
233           2154  29.06.17                       Konica Colour Lab   
235           2155  29.06.17                    Sundaram Clayton Ltd   
237           2156  29.06.17  Vivekanada Institute of Bio Technology   
239           2157  30.06.17                      Mission Technology   
241           2158  30.06.17              DAE Seung Autoparts India    
245           2159  30.06.17                  Fedby Information Ser    
246            NaN       NaN                                     NaN   
247            NaN       NaN                                     NaN   
249           2160  30.06.17              DAE Seung Autoparts India    
252           2161  30.06.17            Integrated Enterprises India   
253            NaN       NaN                                     NaN   
255           2162  30.06.17                      Mission Technology   
257           2163  30.06.17              DAE Seung Autoparts India    
259           2164  30.06.17                Altron Power Equipments    
260            NaN       NaN                                     NaN   
262           2165  30.06.17                

In [ ]:
orders.info()

In [49]:
orders.describe()

Invoice_number      Date Customer_name   Place           Product  \
count               95        95           101     102               150   
unique              95        29            78      68                99   
top               2169  30.06.17   MM Forgings  Guindy  Eaton 9145 1 UPS   
freq                 1        14             8       8                12   

       Quantity  Contact_number  
count       143              79  
unique       37              64  
top         1No      9940915004  
freq         51               4

The above description says : 

   * Most active day of the month : ** 30th June 2017, 14 items. **
   * Most active client of the month : ** MM Forgings, 8 times **
   * Maximum orders came from : ** Guindy, Chennai , 8 times **
   * Best selling product of the month : ** Eaton 9145 1 UPS, 12 times **
   * Total Sales of the month : ** 95 Invoices generated **

In [115]:
# for mailing purpose taking the essential fields alone
essentials = orders[['Date', 'Customer_name', 'Contact_number']].copy()
# .copy() to avoid SettingWithCopyWarning
essentials.dropna(how = 'any',inplace = True)
# drops the row even if one of the field is missing 
essentials.sort_values('Date', inplace = True)
# Sorts the date string

In [ ]:
essentials.info()

In [ ]:
# for eliminating telephone numbers where messages can't be sent
rejected = []
for number in essentials['Contact_number']:
    if len(str(number)) != 10:
        rejected.append(number)
rejected

In [ ]:
# Converting every element in the df into string dtype
essentials.applymap(str)

In [116]:
# for getting the contact details of the customer_name along with the date of purchase
# also resetting the index into 0, 1, 2
essentials.reset_index(drop=True, inplace = True)
essentials

Date                           Customer_name   Contact_number
0   01.06.17                            USUI Susira        9710013151
1   01.06.17                            USUI Susira        9710013151
2   01.06.17                            USUI Susira        9710013151
3   02.06.17                             Hueco India       9600073851
4   03.06.17                              Park Plaza       9382244150
5   03.06.17                           The Residency     0422-2263263
6   03.06.17                             MM Forgings      98409840468
7   05.06.17                             MM Forgings         22352518
8   05.06.17                             MM Forgings       9840980468
9   05.06.17                           Valli Textile  04562-284615/28
10  06.06.17                              VK Digital       9944482786
11  06.06.17                              VK Digital       9944482786
12  06.06.17                           Goodwin Pumps       8754010828
13  06.06.17                         Tekvizion Voice       9940915004
14  06.06.17                               Copy Shop      45566622/44
15  06.06.17                         Tekvizion Voice       9940915004
16  07.06.17                           The Residency     0422-2263263
17  07.06.17                             Tafe Access         27498145
18  08.06.17                          Standyne India       9841420966
19  08.06.17                        K.V.Transmission       9940197844
20  09.06.17                      Mr.M.P.Janarthanam       9840980468
21  09.06.17                     Sysveda Information       9841076572
22  10.06.17                             Check Point       9840773959
23  10.06.17                              KGR Fabtek         64606044
24  12.06.17                        Sundaram Clayton       7550026922
25  12.06.17                             Global Wind       9790925252
26  15.06.17                         Tekvision Voice       9940915004
27  15.06.17                         India Japan IJL         39106000
28  15.06.17                      NSP Infrastructure       8124444724
29  15.06.17                      Advanced Photonics       8547566261
..       ...                                     ...              ...
46  23.06.17                     Sapthrishi Buildcon         28256080
47  23.06.17                  Global Wind Infrastruc       9790925252
48  23.06.17                             Infant Engg         27156341
49  23.06.17                           Fujitec India       9611186843
50  24.06.17                             Galaxy Info       9895522334
51  24.06.17                       Speciality Dental       9445200777
52  24.06.17                      Rajamma Engg P Ltd       8667229788
53  24.06.17                              VK Digital       9500177703
54  26.06.17                           Nuclear Power       9444292810
55  27.06.17                         RS Technologies       9840009016
56  27.06.17                         Neel Industries       9884578008
57  27.06.17                       New Tech Auto Com       8939616592
58  28.06.17                           Swan Electric         28250270
59  29.06.17                    Sundaram Clayton Ltd       9840967320
60  29.06.17  Vivekanada Institute of Bio Technology       9474129451
61  29.06.17                       Konica Colour Lab       9841416519
62  29.06.17                               Medopharm       9551016009
63  29.06.17                   Swaminathan Enterprie       7550097670
64  30.06.17                         Tekvision Voice       9940915004
65  30.06.17                   K.S Engineering Tools       9884989668
66  30.06.17                      Mission Technology       9840033007
67  30.06.17              DAE Seung Autoparts India        9884387786
68  30.06.17                  Fedby Information Ser          49095500
69  30.06.17              DAE Seung Autoparts India        9884387786
70  30.06.17                      Mission Technology       9840033007
71  30.06.17      

In [117]:
# exporting the above dataframe into a csv file named 'resq-June-2017-phone-numbers.csv'
essentials.to_csv('resq-June-2017-phone-numbers.csv', index = True,  index_label = 'Serial', encoding='utf-8')

In [108]:
# for getting mobile numbers alone, which are of 10 digits
essentials = essentials[essentials['Contact_number'].map(lambda x:len(str(x))) == 10]

In [111]:
essentials.reset_index(drop=True, inplace = True)
essentials

Date                           Customer_name Contact_number
0   01.06.17                            USUI Susira      9710013151
1   01.06.17                            USUI Susira      9710013151
2   01.06.17                            USUI Susira      9710013151
3   02.06.17                             Hueco India     9600073851
4   03.06.17                              Park Plaza     9382244150
5   05.06.17                             MM Forgings     9840980468
6   06.06.17                              VK Digital     9944482786
7   06.06.17                              VK Digital     9944482786
8   06.06.17                           Goodwin Pumps     8754010828
9   06.06.17                         Tekvizion Voice     9940915004
10  06.06.17                         Tekvizion Voice     9940915004
11  08.06.17                          Standyne India     9841420966
12  08.06.17                        K.V.Transmission     9940197844
13  09.06.17                      Mr.M.P.Janarthanam     9840980468
14  09.06.17                     Sysveda Information     9841076572
15  10.06.17                             Check Point     9840773959
16  12.06.17                        Sundaram Clayton     7550026922
17  12.06.17                             Global Wind     9790925252
18  15.06.17                         Tekvision Voice     9940915004
19  15.06.17                      NSP Infrastructure     8124444724
20  15.06.17                      Advanced Photonics     8547566261
21  16.06.17                             Mr.Lokesh.A     9444365316
22  16.06.17                             Global Wind     9790925252
23  20.06.17                     Mr.Sankar Narayanan     9840196660
24  20.06.17                             MM Forgings     9840980468
25  20.06.17                             MM Forgings     8754487970
26  20.06.17                             MM Forgings     9840980468
27  21.06.17                           TVS Srichakra     9600990449
28  21.06.17                             Vidyapeetam     9943005792
29  21.06.17                  Shrli Heart & Diabetes     9500191090
..       ...                                     ...            ...
31  22.06.17                       Sathya Components     9962938132
32  22.06.17                      Mr.Vinayagamoorthy     9444441350
33  22.06.17                         Are Pee Polymer     9842991411
34  23.06.17                               Wave Tech     9884298007
35  23.06.17                  Global Wind Infrastruc     9790925252
36  23.06.17                           Fujitec India     9611186843
37  24.06.17                             Galaxy Info     9895522334
38  24.06.17                       Speciality Dental     9445200777
39  24.06.17                      Rajamma Engg P Ltd     8667229788
40  24.06.17                              VK Digital     9500177703
41  26.06.17                           Nuclear Power     9444292810
42  27.06.17                         RS Technologies     9840009016
43  27.06.17                         Neel Industries     9884578008
44  27.06.17                       New Tech Auto Com     8939616592
45  29.06.17                    Sundaram Clayton Ltd     9840967320
46  29.06.17  Vivekanada Institute of Bio Technology     9474129451
47  29.06.17                       Konica Colour Lab     9841416519
48  29.06.17                               Medopharm     9551016009
49  29.06.17                   Swaminathan Enterprie     7550097670
50  30.06.17                         Tekvision Voice     9940915004
51  30.06.17                   K.S Engineering Tools     9884989668
52  30.06.17                      Mission Technology     9840033007
53  30.06.17              DAE Seung Autoparts India      9884387786
54  30.06.17              DAE Seung Autoparts India      9884387786
55  30.06.17                      Mission Technology     9840033007
56  30.06.17              DAE Seung Autoparts India      9884387786
57  30.06.17                Altron Power Equipments      9840423

In [112]:
essentials.to_csv('resq-June-2017.csv', index = True,  index_label = 'Serial', encoding='utf-8')

In [200]:
df = pd.read_csv('resq-June-2017-phone-numbers.csv', index_col = 0)
df

Date                           Customer_name   Contact_number
Serial                                                                   
0       01.06.17                            USUI Susira        9710013151
1       01.06.17                            USUI Susira        9710013151
2       01.06.17                            USUI Susira        9710013151
3       02.06.17                             Hueco India       9600073851
4       03.06.17                              Park Plaza       9382244150
5       03.06.17                           The Residency     0422-2263263
6       03.06.17                             MM Forgings      98409840468
7       05.06.17                             MM Forgings         22352518
8       05.06.17                             MM Forgings       9840980468
9       05.06.17                           Valli Textile  04562-284615/28
10      06.06.17                              VK Digital       9944482786
11      06.06.17                              VK Digital       9944482786
12      06.06.17                           Goodwin Pumps       8754010828
13      06.06.17                         Tekvizion Voice       9940915004
14      06.06.17                               Copy Shop      45566622/44
15      06.06.17                         Tekvizion Voice       9940915004
16      07.06.17                           The Residency     0422-2263263
17      07.06.17                             Tafe Access         27498145
18      08.06.17                          Standyne India       9841420966
19      08.06.17                        K.V.Transmission       9940197844
20      09.06.17                      Mr.M.P.Janarthanam       9840980468
21      09.06.17                     Sysveda Information       9841076572
22      10.06.17                             Check Point       9840773959
23      10.06.17                              KGR Fabtek         64606044
24      12.06.17                        Sundaram Clayton       7550026922
25      12.06.17                             Global Wind       9790925252
26      15.06.17                         Tekvision Voice       9940915004
27      15.06.17                         India Japan IJL         39106000
28      15.06.17                      NSP Infrastructure       8124444724
29      15.06.17                      Advanced Photonics       8547566261
...          ...                                     ...              ...
46      23.06.17                     Sapthrishi Buildcon         28256080
47      23.06.17                  Global Wind Infrastruc       9790925252
48      23.06.17                             Infant Engg         27156341
49      23.06.17                           Fujitec India       9611186843
50      24.06.17                             Galaxy Info       9895522334
51      24.06.17                       Speciality Dental       9445200777
52      24.06.17                      Rajamma Engg P Ltd       8667229788
53      24.06.17                              VK Digital       9500177703
54      26.06.17                           Nuclear Power       9444292810
55      27.06.17                         RS Technologies       9840009016
56      28.07.16                         Neel Industries       9884578008
57      28.07.16                       New Tech Auto Com       8939616592
58      28.06.17                           Swan Electric         28250270
59      29.06.17                    Sundaram Clayton Ltd       9840967320
60      29.06.17  Vivekanada Institute of Bio Technology       9474129451
61      29.06.17                       Konica Colour Lab       9841416519
62      29.06.17                               Medopharm       9551016009
63      29.06.17                   Swaminathan Enterprie       7550097670
64      30.06.17                         Tekvision Voice       9940915004
65      30.06.17                   K.S Engineering Tools       9884989668
66      30.06.17                      Mission Technology       9840033007
67      30.06.17

In [257]:
# to get the one year old dates from the csv and send email to the admin, to renew the client's product 

current_date = datetime.now().strftime("%d.%m.%y")
to_check = current_date[:-2]
df = df.loc[df['Date'] == to_check + str(int(current_date[-2:]) - 1), :]

if df.empty:
    msg = 'No renewals today'
else:
    df.reset_index(drop=True, inplace = True)
    contact_clients = []
    for index, row in df.iterrows():
         contact_clients.append(str(index) + ' ' + str(row['Date']) + ' ' + str(row['Customer_name']) + ' ' + str(row['Contact_number']))
    print(contact_clients)
    

['0 28.07.16 Neel Industries 9884578008', '1 28.07.16 New Tech Auto Com 8939616592']


In [262]:
# for sending the email to admin of resq
# note : had to 'allow permission to less secure apps', in gmail else you would get 'SMTPAuthenticationError', though your email id and password being correct

import smtplib
smtpObj = smtplib.SMTP('smtp.gmail.com', 587)
smtpObj.ehlo()
smtpObj.starttls()
smtpObj.login("jaiimmortal@gmail.com", "NoWayThisIsMyPwd6969")
if df.empty:
    msg = 'No renewals today'
    smtpObj.sendmail("jaiimmortal@gmail.com", "yoyaswant@gmail.com", 'Subject: {}\n'.format(current_date[:-3]) + msg)
else:
    s = '\n'
    msg = s.join(contact_clients)
    smtpObj.sendmail("jaiimmortal@gmail.com", "yoyaswant@gmail.com", 'Subject: {}\n'.format(current_date[:-3]) + msg)
smtpObj.quit()


(221, b'2.0.0 closing connection z16sm36640117pgc.90 - gsmtp')